Common/Serialize/Serializer.h

```diff
@@ -192,47 +192,63 @@ class CChunkFileReader
 		return (size_t)ptr;
 	}
 
-	// Expects ptr to have at least MeasurePtr bytes at ptr.
+	// If *saved is null, will allocate storage using malloc.
+	// If it's not null, it will be used, but only hope can save you from overruns at the end. For libretro.
 	template<class T>
-	static Error SavePtr(u8 *ptr, T &_class, size_t expected_size)
+	static Error MeasureAndSavePtr(T &_class, u8 **saved, size_t *savedSize)
 	{
-		const u8 *expected_end = ptr + expected_size;
-		PointerWrap p(&ptr, PointerWrap::MODE_WRITE);
+		u8 *ptr = nullptr;
+		PointerWrap p(&ptr, PointerWrap::MODE_MEASURE);
+		_class.DoState(p);
+		_assert_(p.error == PointerWrap::ERROR_NONE);
+
+		size_t measuredSize = p.Offset();
+		u8 *data;
+		if (*saved) {
+			data = *saved;
+		} else {
+			data = (u8 *)malloc(measuredSize);
+			if (!data)
+				return ERROR_BAD_ALLOC;
+		}
+
+		p.RewindForWrite(data);
 		_class.DoState(p);
 
-		if (p.error != PointerWrap::ERROR_FAILURE && (expected_end == ptr || expected_size == 0)) {
```

Comment:
It might still be nice to validate that the size we got to matched measuredSize, we've had bugs in the past so it's nice if it trips.

-[Unknown]

Reply:
This is done internally in CheckAfterWrite.